### PR TITLE
Add flatpak firststart wizard

### PR DESCRIPTION
**Pull request by carstene1ns**
This is one solution to the startup folder problem our flatpak installation has.

This writes an additional config file, that is sourced by the startup script.

The script allows command line overwrite of game directory and launching the wizard.

### CODE REVIEW

File: Flatpak/firststart/src/main.c [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+		sprintf(filename, "%s/%s", directory, "flatpak.sh");
+		printf("Saving to: %s\n", filename);
+		FILE *file = fopen(filename, "w");
+		fputs(prelude, file);
+		
+		// only option for now
+		char *startupdir = gtk_file_chooser_get_filename(dirchooser);
+		if(startupdir != NULL) {
+			fputs("# Startup directory (for game browser)\n", file);
+			fprintf(file, "EASYRPG_STARTUP_DIR=%s\n", startupdir);
```
Comment (Ghabry) [inline Flatpak/firststart/src/main.c]: Could be ``RPG_GAME_PATH``. That var is used by the Player already.

File: Flatpak/firststart/ui/assistant.glade [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  <object class="GtkTextBuffer" id="textbuffer_start">
+    <property name="text" translatable="yes">EasyRPG Player will usually start from the current
+working directory. However, this is not too useful
+in a Flatpak environment.
+
+Therefore this wizard lets you choose a different
+startup folder.
+
+Navigate to your game directory
+(e.g. /home/user/Downloads/RPG) and save.</property>
```
Comment (Ghabry) [inline Flatpak/firststart/ui/assistant.glade]: Imo the text should be more user agnostic and not explain the technical reason, e.g.:

> This wizard lets you choose the game directory of EasyRPG Player.
> 
> Navigate to your game directory containing RPG Maker 2000/2003 games
(e.g. /home/user/Downloads/RPG) and save.

File: Flatpak/firststart/src/main.c [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+		if(tmp == NULL) {
+			// non-flatpak fallback
+			tmp = getenv("HOME");
+			sprintf(config_home, "%s/%s", tmp, ".config");
+		} else {
+			strcpy(config_home, tmp);
+		}
+		char directory[120];
+		sprintf(directory, "%s/%s", config_home, "EasyRPG/Player");
+		mkdir(directory, 0700);
```
Comment (Ghabry) [inline Flatpak/firststart/src/main.c]: ``mkdir`` must be invoked in a loop to create all parent directories

**Review by Ghabry — COMMENTED** [review]
Thanks for the nice ui! This is better than a simple "Make this the startup dir" setting in the game browser.

File: Flatpak/firststart/src/main.c (hunk 1 above)
Comment (carstene1ns) [reply Flatpak/firststart/src/main.c]: oh right,  good idea

**Comment by Ghabry** [comment]
Is still a draft but for me (and my limited GTK exp) the current state looks fine.

**Comment by carstene1ns** [comment]
Now with launcher script.

Missing is adding this to the flatpak build manifests.

**Review by Ghabry — APPROVED** [review]
Did various invocations of the script. Passed all my testing.

**Pull request merged**